pkg/client/events/replay_client.go

```diff
@@ -84,48 +85,74 @@ type replayClient[T any, U observable.ReplayObservable[T]] struct {
 //
 // Required dependencies:
 //   - client.EventsQueryClient
-func NewEventsReplayClient[T any, U observable.ReplayObservable[T]](
+func NewEventsReplayClient[T any, R observable.ReplayObservable[T]](
 	ctx context.Context,
 	deps depinject.Config,
-	cometWebsocketURL string,
 	queryString string,
 	newEventFn NewEventsFn[T],
 	replayObsBufferSize int,
-) (client.EventsReplayClient[T, U], error) {
+) (client.EventsReplayClient[T, R], error) {
 	// Initialize the replay client
-	rClient := &replayClient[T, U]{
-		endpointURL:  cometWebsocketURL,
-		queryString:  queryString,
-		eventDecoder: newEventFn,
+	rClient := &replayClient[T, R]{
+		queryString:         queryString,
+		eventDecoder:        newEventFn,
+		replayObsBufferSize: replayObsBufferSize,
 	}
-	replayObsCache, replayObsCachePublishCh := channel.NewReplayObservable[U](
+	// TODO_REFACTOR(@h5law): Look into making this a regular observable
+	replayObsCache, replayObsCachePublishCh := channel.NewReplayObservable[R](
 		ctx,
-		replayObsBufferSize,
+		// buffer size of 1 as the cache only needs to hold the latest
```

Comment:
Yes, it was incorrect previously. In the current construction (i.e. the cache is a replay observable), we only ever need to replay the most recent value from the cache because older values are references observables which have been abandoned (i.e. will never be notified again) due to network interruption.

Since this PR refactors `#EventSequence()` to no longer depend on `replayObsCache.Last()`, it need not be a replay observable.

Reply:
So when we have a replay observable of observables, is there a use case of it being greater than one?

(no changes necessary, just trying to understand)

Reply:
Correct, the idea was that we could just always have the most up-to-date reference to the "active" event data stream by calling `#Last()`. However, what we discovered is that there's a race between the caller and when the observable is notified with the new reference. This is the reason that `EventsReplayClient#EventsSequence()` now composes and returns a new observable instead, it perfectly synchronizes where the race was before.

As we're no longer using the replay-specific API (i.e. `#Last()`), It doesn't matter anymore and we should be able to simplify this again in a future iteration.